examples/quickstart-xgboost/client.py

```diff
@@ -0,0 +1,150 @@
+import warnings
+from logging import INFO
+import xgboost as xgb
+
+import flwr as fl
+from flwr_datasets import FederatedDataset
+from flwr.common.logger import log
+from flwr.common import (
+    Code,
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    GetParametersIns,
+    GetParametersRes,
+    Parameters,
+    Status,
+)
+
+from dataset import instantiate_partitioner, train_test_split, transform_dataset_to_dmatrix
+
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+# Load (HIGGS) dataset and conduct partitioning
+num_partitions = 20
+# partitioner type is chosen from ["uniform", "linear", "square", "exponential"]
+partitioner_type = "uniform"
+
+# instantiate partitioner
+partitioner = instantiate_partitioner(partitioner_type=partitioner_type, num_partitions=num_partitions)
+fds = FederatedDataset(dataset="jxie/higgs", partitioners={"train": partitioner})
+
+# let's use the first partition as an example
+partition_id = 0
```

Comment:
That has to be changed (each client now has exactly the same dataset). I'd propose we have argparse and the node is specified from the command line.

Reply:
The argparse is exactly what i proposed for the next PR, which allows users to select different experimental settings. In this PR, I just follow `quickstart-pytorch` and did the same thing. For the demo purpose, i think it's fine using the same data for all clients (same as `quickstart-pytorch`), but i agree that it'd be better to provide more realistic experiments.

I don't mind adding argparse stuffs in this PR.

Reply:
I'd propose to add it here, but maybe let's ask Daniel for feedback. (also centralized evaluation is not really an advanced feature according to me)
The advanced Pytroch basically adds config, which is already in this PR. 
Otherwise, the instantiation of Partitioner should be dropped, and instead of having a whole function with choices, one should be instantiated since the support for others is not present.